### ISSUE TITLE
[BUG] Unable to query OneToMany/ManyToOne relations

### ISSUE THREAD

**Issue by jjdrake** [laravel-doctrine/orm#341]
I am using php v7.2.x, Laravel-Doctrine v1.4.7 with Lumen v5.6 (as well as v5.5 for compatibility with listed supported Laravel versions).

I've set up the following simplified entities which establish OneToMany & ManyToOne relationships:

```
use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;

/**
 * @ORM\Table(name="Menu")
 * @ORM\Entity
 */
class Menu
{
    /**
     * @ORM\OneToMany(targetEntity="Domain\MenuItem", mappedBy="menu", cascade={"persist"})
     */
    private $items;

    public function __construct()
    {
        $this->items = new ArrayCollection();
    }
}
```

```
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Table(name="MenuItem")
 * @ORM\Entity
 */
class MenuItem
{
    /**
     * @ORM\ManyToOne(targetEntity="Domain\Menu", inversedBy="items")
     * @ORM\JoinColumn(name="menuId", referenceColumnName="id")
     */
    private $menu;
}
```

I'm able to perform inserts with this set-up without issue.  However, when selecting menus via the repository find method I wasn't receiving any results for items, (despite items existing in the database).  When I attempted to dump out the contents of $items it would hang for a bit and ultimately crash Postman, etc.  The furthest I could track it down was to the hydrateAdd method of Doctrine's PersistentCollection class.  There appeared to be an issue retrieving the owner entity of the given collection.  Attempting to var_dump the owner would again simply crash whatever program I was using to make the calls.  Incidentally, I ran an audit log on my underlying MySQL database and Doctrine's query used to perform the selection appeared fine and was generating the correct results.  If I remove this association I am able to successfully query either table.

**Comment by eigan** [comment]
Have you tried to validate the schema? Looks like you could add inversedBy at least

**Comment by jjdrake** [comment]
@eigan Yes, I've validated the schema but fair point about adding the inversedBy tag;  I actually had the tag in place but didn't copy it over correctly in the comment, (I've updated the comment to better reflect it now).

Some additional info I've learned since my original comment:  I am able to successfully perform a count on the items, (which accurately reflects the number of records in the database).  I'm just not able to dump them out or get their values; attempting so will ultimately crash whatever program I use to make the call.  After removing the association from the Menu entity and dumping out the items by themselves I did notice messages about recursion in the output.  It seems that the fact that the Menu entity contains a list of MenuItems which in turn contain a reference to the same Menu is likely causing an issue.  My code does appear to adhere to documentation though.  Isn't Doctrine prepared to handle this type of circular reference internally?
